lib/puppet/provider/docker_compose/ruby.rb

```diff
@@ -0,0 +1,29 @@
+Puppet::Type.type(:docker_compose).provide(:ruby) do
+  desc "Support for Docker Compose"
+  commands :dockercompose => "docker-compose"
+    
+  mk_resource_methods
+
+
+  def exists?
+    Puppet.info("Checking if docker-compose.yml exists")
+    begin
+      Dir.chdir(resource[:source])
+      File.file?('docker-compose.*')    
+    rescue Exception => e
+       false 
+    end
+  end
+ 
+  def create
+     Puppet.info("running docker-compose up -d")
+     Dir.chdir(resource[:source])
+     dockercompose('up', '-d')
+  end
+   
+  def destroy
+    Puppet.info("stoping docker-compose containers")
+    Dir.chdir(resource[:source])
+    dockercompose('stop')
```

Comment:
stop will stop the containers from running but not remove them. I'm not sure the semantics are quite right here. I think if you are only supporting absent and present for ensure to begin with then absent should remove the containers completely.